arch/xtensa/core/xtensa-asm2-util.S

```diff
@@ -147,35 +147,63 @@ _high_restore_done:
  * though...
  */
 
-#define RCTX_EPC_REG		_CONCAT(EPC, XCHAL_NUM_INTLEVELS)
-#define RCTX_EPS_REG		_CONCAT(EPS, XCHAL_NUM_INTLEVELS)
+/*
+ * XCC doesn't like _CONCAT() so we need to spell out
+ * every single one.
```

Comment:
Hm... is this something to do with the clang preprocessor?  Seems like we're going to hit this elsewhere if _CONCAT doesn't work on clang.

Reply:
This seems to be only limited to the assembler. We have C code using _CONCAT and XCC-Clang never complains.